R/zchunk_L244.building_USA.R

```diff
@@ -0,0 +1,917 @@
+      satiation.adder = if_else(satiation.adder > pcFlsp_mm2, pcFlsp_mm2 * 0.999, satiation.adder)) %>%
+        select(LEVEL2_DATA_NAMES[["SatiationAdder"]])
+
+    # Heating and cooling degree days (thermal services only)
+    # First, separate the thermal from the generic services. Generic services will be assumed to produce
+    # internal gain energy, so anything in the internal gains assumptions table will be assumed generic
+    generic_services <- unique(A44.globaltech_intgains$supplysector)
+    thermal_services <- setdiff(unique(A44.sector$supplysector), generic_services)
+
+    # L244.HDDCDD: Heating and cooling degree days by scenario
+    L244.HDDCDD_scen_state <- L143.HDDCDD_scen_state %>%
+      rename(region = state,
+             degree.days = value)
+
+    # Let's make a climate normal (historical average) for each region, using a selected interval of years
+    # Don't want to just set one year, because we want average values for all regions
+    L244.HDDCDD_normal_state <- L244.HDDCDD_scen_state %>%
+      filter(year %in% seq(1981, 2000)) %>%
+      group_by(region, variable) %>%
+      summarise(degree.days = mean(degree.days)) %>%
+      ungroup()
+
+    # Subset the heating and cooling services, separately
+    heating_services <- thermal_services[grepl("heating", thermal_services)]
+    cooling_services <- thermal_services[grepl("cooling", thermal_services)]
+
+    L244.HDDCDD_A2_GFDL <- tidyr::crossing(region = gcamusa.STATES, thermal.building.service.input = thermal_services) %>%
+      # Add in gcam.consumer
+      left_join_error_no_match(calibrated_techs_bld_usa %>%
+                                 select(service, gcam.consumer = sector) %>%
+                                 distinct(), by = c("thermal.building.service.input" = "service")) %>%
+      # Add in nodeInput and building.node.input
+      left_join_error_no_match(A44.gcam_consumer, by = "gcam.consumer") %>%
+      select(LEVEL2_DATA_NAMES[["BldNodes"]], thermal.building.service.input) %>%
+      # Add in model years
+      repeat_add_columns(tibble(year = MODEL_YEARS)) %>%
+      # Add HDD/CDD so that we can join with L244.HDDCDD_scen_state, remove at end
+      mutate(variable = if_else(thermal.building.service.input %in% heating_services, "HDD", "CDD")) %>%
+      # Add in degree days
+      left_join_error_no_match(L244.HDDCDD_scen_state, by = c("region", "variable", "year")) %>%
+      mutate(degree.days = round(degree.days, energy.DIGITS_HDDCDD)) %>%
+      # Don't need to keep Scen and GCM identifiers because only one is used
+      select(-Scen, -GCM, -variable)
+
+    # L244.ShellConductance_bld_gcamusa: Shell conductance (inverse of shell efficiency)
+    L244.ShellConductance_bld_gcamusa <- A44.bld_shell_conductance %>%
+      # Convert to long form
+      gather(year, value, matches(YEAR_PATTERN)) %>%
+      mutate(year = as.integer(year)) %>%
```

Comment:
Replace these new lines with call to new `gather_year` function (merged in #821 )